CommandLine.c

```diff
@@ -204,12 +205,15 @@ static CommandLineStatus parseArguments(int argc, char** argv, CommandLineSettin
             if (!username) {
                flags->userId = geteuid();
             } else if (!Action_setUserOnly(username, &(flags->userId))) {
-               for (const char* itr = username; *itr; ++itr)
-                  if (!isdigit((unsigned char)*itr)) {
+               char *endptr;
+               errno = 0;
+               unsigned long res = strtoul(username, &endptr, 10);
+               unsigned castRes = (unsigned) res;
+               if (*endptr != '\0' || res == ULONG_MAX || errno != 0 || castRes != res) {
```

Comment:
Before I can suggest a better code for this, I need one behavior clarified:
Is `username` being an empty string acceptable here?
Because `strtoul` would not produce an error for the empty string. It would happily return 0 as the result.